File: src/main/java/io/github/foodjournal/domain/QuantityUnit.java
package io.github.foodjournal.domain;

import java.util.Locale;

public enum QuantityUnit {
  G("g"), ML("ml"), PORTION("portion"), UNSPECIFIED("unspecified");

  private final String databaseValue;
  QuantityUnit(String databaseValue){this.databaseValue=databaseValue;}
  public String databaseValue(){return databaseValue;}
  public static QuantityUnit fromDatabaseValue(String value){
    if(value==null||value.isBlank())return UNSPECIFIED;
    return switch(value.toLowerCase(Locale.ROOT)){
      case "g" -> G; case "ml" -> ML; case "portion" -> PORTION; case "unspecified" -> UNSPECIFIED;
      default -> throw new IllegalArgumentException("Unsupported quantity unit: "+value);
    };
  }
}

File: python-app/app/domain/quantity_unit.py
from enum import Enum


class QuantityUnit(str, Enum):
    G = "g"
    ML = "ml"
    PORTION = "portion"
    UNSPECIFIED = "unspecified"

    @classmethod
    def from_database_value(cls, value: str | None) -> "QuantityUnit":
        if value is None or not value.strip():
            return cls.UNSPECIFIED
        try:
            return cls(value.strip())
        except ValueError as exc:
            raise ValueError(f"Unrecognized quantity unit: {value}") from exc
